src/vdaf/prio3.rs

```diff
@@ -1401,6 +1401,57 @@ where
     }
 }
 
+/// This is a polyfill for `usize::ilog2()`, which is only available in Rust 1.67 and later. It is
+/// based on the implementation in the standard library. It can be removed when the MSRV has been
+/// advanced past 1.67.
+///
+/// # Panics
+///
+/// This function will panic if `input` is zero.
+fn ilog2(input: usize) -> u32 {
+    if input == 0 {
+        panic!("Tried to take the logarithm of zero");
+    }
+    (usize::BITS - 1) - input.leading_zeros()
+}
+
+/// Finds the optimal choice of chunk length for [`Prio3Histogram`] or [`Prio3SumVec`], given its
+/// encoded measurement length. For [`Prio3Histogram`], the measurement length is equal to the
+/// length parameter. For [`Prio3SumVec`], the measurement length is equal to the product of the
+/// length and bits parameters.
+pub fn optimal_chunk_length(measurement_length: usize) -> usize {
```

Comment:
Maybe nitpicky--can we have constructors for Prio3Histogram and Prio3SumVec that do the optimal calculation for us?

e.g.
```rust
impl Prio3SumVec {
    pub fn new_sum_vec_with_optimal_chunk_size( // feel free to editorialize this function name
        num_aggregators: u8,
        bits: usize,
        len: usize,
    ) -> Result<Self, VdafError> {
        Prio3::new(
            num_aggregators,
            SumVec::new(bits, len, optimal_chunk_length(bits * len))?,
        )
    }
}
```

Same for Prio3Histogram except call would be `optimal_chunk_length(len)`. Prevents the developer from making a mistake in calculating the measurement size.

Reply:
There are some cases where there are multiple equally-optimal chunk lengths that produce the same proof length. If another implementation made a different choice in such a case, or if we changed this search algorithm in the future, then we'd run the risk of different parties disagreeing on the chunk length, which would lead to proof verification failures. So long as the chunk length is still a VDAF parameter, I think we shouldn't encourage selecting the parameter in more than one place by bundling this logic into the VDAF constructors.